core/arch/arm/kernel/user_ta.c

```diff
@@ -591,25 +592,44 @@ static const struct tee_ta_ops user_ta_ops __rodata_unpaged = {
 	.get_instance_id = user_ta_get_instance_id,
 };
 
-static const struct user_ta_store_ops *user_ta_store;
+static SLIST_HEAD(uta_stores_head, user_ta_store_ops) uta_store_list =
+		SLIST_HEAD_INITIALIZER(uta_stores_head);
 
-TEE_Result tee_ta_register_ta_store(const struct user_ta_store_ops *ops)
+TEE_Result tee_ta_register_ta_store(struct user_ta_store_ops *ops)
 {
-	user_ta_store = ops;
+	struct user_ta_store_ops *p = NULL;
+	struct user_ta_store_ops *e;
+
+	DMSG("Registering TA store: '%s' (priority %d)", ops->description,
+	     ops->priority);
+
+	SLIST_FOREACH(e, &uta_store_list, link) {
+		if (e->priority > ops->priority)
```

Comment:
I thought I would allow equal priorities, but it may be a bad idea (invocation order would depend on registration order which may not be well controlled...). I will add the assert.